libvips/foreign/fits.c

```diff
@@ -139,7 +139,8 @@ static void
 vips_fits_close(VipsFits *fits)
 {
 	VIPS_FREE(fits->filename);
-	VIPS_FREEF(vips_g_mutex_free, fits->lock);
+	if (fits->line)
+		g_mutex_clear(&fits->lock);
```

Comment:
This isn't ideal. An alternative might be merging this with `fitsload.c`, but that is a much larger change.